synapse/cortex.py

```diff
@@ -5754,10 +5758,30 @@ async def _getMirrorProxy(self):
                 # we are part of the pool and were selected. Convert to local use.
                 return None
 
+            curoffs = await self.getNexsIndx()
+            miroffs = await s_common.wait_for(proxy.getNexsIndx(), timeout)
+            if (delta := curoffs - miroffs) > MAX_NEXUS_DELTA:
+                extra = {
+                    'delta': delta,
+                    'mirror': proxyname,
+                    'mirror_offset': miroffs,
+                }
+                mesg = (f'Pool mirror [{proxyname}] Nexus offset delta too large '
+                        f'({delta} > {MAX_NEXUS_DELTA}), running query locally.')
+                logger.warning(mesg, extra={'synapse': extra})
+                return None
+
+            opts.setdefault('nexsoffs', curoffs - 1)
+
             return proxy
 
         except (TimeoutError, s_exc.IsFini):
-            logger.warning('Timeout waiting for pool mirror, running query locally.')
+            if proxy is None:
+                logger.warning('Timeout waiting for pool mirror, running query locally.')
+            else:
+                mesg = f'Timeout waiting for pool mirror [{proxyname}] Nexus offset, running query locally.'
+                logger.warning(mesg, extra={'synapse': {'mirror': proxyname}})
```

Comment:
`self.cell.getLogExtra(mirror=proxyname)` could be used here too if we wanted to record information about the telepath caller.